internal/otelcollector/config/metricgateway/config_builder.go

```diff
@@ -108,40 +82,262 @@ func (b *Builder) Build(ctx context.Context, pipelines []telemetryv1alpha1.Metri
+			}
+
+			return filterByNamespaceProcessorConfig(input.OTLP.Namespaces, fmt.Sprintf("not(%s)", common.ScopeNameEquals(common.InstrumentationScopeKyma)))
+		},
+	)
+}
+
+// Resource processors
+
+func (b *Builder) addInsertClusterAttributesProcessor(opts BuildOptions) buildComponentFunc {
+	return b.AddProcessor(
+		b.StaticComponentID(common.ComponentIDInsertClusterAttributesProcessor),
+		func(mp *telemetryv1alpha1.MetricPipeline) any {
+			return common.InsertClusterAttributesProcessorConfig(opts.ClusterName, opts.ClusterUID, opts.CloudProvider)
+		},
+	)
+}
+
+func (b *Builder) addDropKymaAttributesProcessor() buildComponentFunc {
+	return b.AddProcessor(
+		b.StaticComponentID(common.ComponentIDDropKymaAttributesProcessor),
+		func(mp *telemetryv1alpha1.MetricPipeline) any {
+			return common.DropKymaAttributesProcessorConfig()
+		},
+	)
+}
+
+func (b *Builder) addUserDefinedTransformProcessor() buildComponentFunc {
+	return b.AddProcessor(
+		formatUserDefinedTransformProcessorID,
+		func(mp *telemetryv1alpha1.MetricPipeline) any {
+			if len(mp.Spec.Transforms) == 0 {
+				return nil // No transforms, no processor needed
+			}
+
+			transformStatements := common.TransformSpecsToProcessorStatements(mp.Spec.Transforms)
+			transformProcessor := common.MetricTransformProcessorConfig(transformStatements)
+
+			return transformProcessor
+		},
+	)
+}
+
+func (b *Builder) addUserDefinedFilterProcessor() buildComponentFunc {
+	return b.AddProcessor(
+		formatUserDefinedFilterProcessorID,
+		func(mp *telemetryv1alpha1.MetricPipeline) any {
+			if mp.Spec.Filters == nil {
+				return nil // No filters, no processor needed
+			}
+
+			return common.FilterSpecsToMetricFilterProcessorConfig(mp.Spec.Filters)
+		},
+	)
+}
+
+// Batch processor
+
+//nolint:mnd // hardcoded values
+func (b *Builder) addBatchProcessor() buildComponentFunc {
+	return b.AddProcessor(
+		b.StaticComponentID(common.ComponentIDBatchProcessor),
+		func(mp *telemetryv1alpha1.MetricPipeline) any {
+			return &common.BatchProcessor{
+				SendBatchSize:    1024,
+				Timeout:          "10s",
+				SendBatchMaxSize: 1024,
+			}
+		},
+	)
+}
+
+func (b *Builder) addOTLPExporter(queueSize int) buildComponentFunc {
+	return b.AddExporter(
+		formatOTLPExporterID,
+		func(ctx context.Context, mp *telemetryv1alpha1.MetricPipeline) (any, common.EnvVars, error) {
+			otlpExporterBuilder := common.NewOTLPExporterConfigBuilder(
+				b.Reader,
+				mp.Spec.Output.OTLP,
+				mp.Name,
+				queueSize,
+				common.SignalTypeMetric,
+			)
+
+			return otlpExporterBuilder.OTLPExporterConfig(ctx)
+		},
+	)
+}
+
+// Helper functions
+
+func shouldFilterByNamespace(namespaceSelector *telemetryv1alpha1.NamespaceSelector) bool {
+	return namespaceSelector != nil && (len(namespaceSelector.Include) > 0 || len(namespaceSelector.Exclude) > 0)
+}
+
+func filterByNamespaceProcessorConfig(namespaceSelector *telemetryv1alpha1.NamespaceSelector, inputSourceCondition string) *common.FilterProcessor {
```

Comment:
This function used to be used for multiple inputs, now it's only used by the OTLP input. You could simplify it greatly/